src/cabinDatabase/cabinDatabase.api.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+  createClient,
+  PostgrestResponse,
+  SupabaseClient,
+} from '@supabase/supabase-js';
+import { SupabaseCabin } from './cabinDatabase.interface';
+
+@Injectable()
+export class CabinDatabaseApi implements OnModuleInit {
+  private supabase: SupabaseClient;
+  private CABIN_REQUEST_DEFAULT_LIMIT = 1000; // TODO 1000 is an artificial limit to prevent limit undefined
+
+  constructor(private readonly configService: ConfigService) {}
+
+  onModuleInit() {
+    const supabaseUrl = this.configService.getOrThrow('SUPABASE_URL');
+    const supabaseKey = this.configService.getOrThrow('SUPABASE_KEY');
+    this.supabase = createClient(supabaseUrl, supabaseKey);
```

Comment:
I think you can move this to the constructor?

Reply:
True, done.